src/clientAtom.ts

```diff
@@ -1,13 +1,36 @@
-import { InMemoryCache, ApolloClient } from '@apollo/client'
+import {
+  InMemoryCache,
+  ApolloClient,
+  NormalizedCacheObject,
+} from '@apollo/client'
 import { atom } from 'jotai/vanilla'
 
 const DEFAULT_URL =
   (typeof process === 'object' && process.env.JOTAI_APOLLO_DEFAULT_URL) ||
   '/graphql'
 
-const client = new ApolloClient({
-  uri: DEFAULT_URL,
-  cache: new InMemoryCache(),
-})
+let defaultClient: ApolloClient<NormalizedCacheObject> | null = null
 
-export const clientAtom = atom(client)
+const customClientAtom = atom<ApolloClient<unknown> | null>(null)
+
+export const clientAtom = atom(
+  (get) => {
+    const customClient = get(customClientAtom)
+
+    if (!customClient) {
+      return customClient
+    }
+
+    if (!defaultClient) {
+      defaultClient = new ApolloClient({
+        uri: DEFAULT_URL,
+        cache: new InMemoryCache(),
+      })
+    }
+
+    return defaultClient
+  },
+  (_get, set, client: ApolloClient<unknown>) => {
+    set(customClientAtom, client)
```

Comment:
Oh, cool. I thought it wouldn't work first, but yeah this should work. (It didn't work in v1, but v2 should be fine.)